merge_condensed_sdrfs.py

```diff
@@ -100,49 +128,73 @@ def cluster_samples(cond: pd, main_covariate: str, covariate_type: str,
                     G.edges[batches[i], batches[j]]['covariate'].update([cov])
                 else:
                     G.add_edge(batches[i], batches[j], covariate=set([cov]))
+
     conn_components = [G.subgraph(c).copy() for c in nx.connected_components(G)]
-    print("Number of connected components: {}".format(str(len(conn_components))))
-    largest_conn_comp = max(conn_components, key=len)
-    print("Largest connected component size: {}".format(str(len(largest_conn_comp))))
-    conn_components.sort(key=len, reverse=True)
-    for cc in conn_components:
-        print("CC number of batches: {}".format(str(len(cc))))
-        print("CC batches: {}".format(cc.nodes))
-        covariates_cc = set()
-        for edge_cov in cc.edges.data('covariate'):
-            covariates_cc.update(edge_cov[2])
-        print("CC number of covariates: {}".format(str(len(covariates_cc))))
-        print("CC covariates: {}".format(covariates_cc))
-        print()
+    if conn_components:
+        print("Number of connected components: {}".format(str(len(conn_components))))
+        largest_conn_comp = max(conn_components, key=len)
+        print("Largest connected component size: {}".format(str(len(largest_conn_comp))))
+        conn_components.sort(key=len, reverse=True)
+        first_cc_covariates = []
+        second_cc_covariates = []
+        for cc in conn_components:
+            print("CC number of batches: {}".format(str(len(cc))))
+            print("CC batches: {}".format(cc.nodes))
+            covariates_cc = set()
+            for edge_cov in cc.edges.data('covariate'):
+                covariates_cc.update(edge_cov[2])
+            if not second_cc_covariates and first_cc_covariates:
+                second_cc_covariates.extend(covariates_cc)
+            if not first_cc_covariates:
+                first_cc_covariates.extend(covariates_cc)
+            print("CC number of covariates: {}".format(str(len(covariates_cc))))
+            print("CC covariates: {}".format(covariates_cc))
+            print()
```

Comment:
I haven't seen the empty print statement used as a line break. `\n` in the format string is probably more explicit.

Reply:
Resolved now!